src/prefect/blocks/core.py

```diff
@@ -660,7 +692,22 @@ class Custom(Block):
                 f"Unable to find block document named {block_document_name} for block type {block_type_slug}"
             ) from e
 
-        return cls._from_block_document(block_document)
+        try:
+            return cls._from_block_document(block_document)
+        except ValidationError as e:
+            if not validate:
+                missing_fields = tuple(err["loc"][0] for err in e.errors())
+                missing_block_data = {field: None for field in missing_fields}
+                warnings.warn(
+                    f"Could not fully load {block_document_name!r} of block type {cls.__name__!r} - "
+                    "this is likely because one or more required fields were added to the schema "
+                    f"for {cls.__name__!r} that did not exist on the class when this block was last saved. "
+                    f"Please specify values for new field(s): {listrepr(missing_fields)}, then "
+                    f'run `{cls.__name__}.save("{block_document_name}", overwrite=True)`, and '
+                    "load this block again before attempting to use it."
+                )
+                return cls.construct(**block_document.data, **missing_block_data)
+            raise
```

Comment:
I think we should log a message `validate=True` and validation fails so that users know how to proceed.
```suggestion
            logger.exception(f"Unable to load {block_document_name!r} of block type {cls.block_type_slug} due to failed validation. To load without validation, trying loading again with `validate=False`.") 
            raise
```

Reply:
is this how we want this message to appear? e.g.

```python
In [1]: from prefect.blocks.core import Block

In [2]: class Foo(Block):
   ...:     x: dict
   ...:

In [3]: Foo.load("foo")
10:41:49.518 | ERROR   | root - Unable to load 'foo' of block type None due to failed validation. To load without validation, try loading again with `validate=False`.
Traceback (most recent call last):
  File "/Users/nate/src/open-source/prefect-stuff/prefect/src/prefect/blocks/core.py", line 697, in load
    return cls._from_block_document(block_document)
  File "/Users/nate/src/open-source/prefect-stuff/prefect/src/prefect/blocks/core.py", line 552, in _from_block_document
    block = block_cls.parse_obj(block_document.data)
  File "pydantic/main.py", line 526, in pydantic.main.BaseModel.parse_obj
  File "/Users/nate/src/open-source/prefect-stuff/prefect/src/prefect/blocks/core.py", line 184, in __init__
    super().__init__(*args, **kwargs)
  File "pydantic/main.py", line 342, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Foo
x
  value is not a valid dict (type=type_error.dict)
-----------------------------------------------------------------------
```

Reply:
switching to raising a `RuntimeError` from the `ValidationError` looks much better
```python

RuntimeError: Unable to load 'foo' of block type 'foo' due to failed validation. To load without validation, try loading again with `validate=False`.

```